coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -210,6 +210,44 @@ func (server *Server) AcquireJob(ctx context.Context, _ *proto.Empty) (*proto.Ac
 			return nil, failJob(fmt.Sprintf("get workspace build parameters: %s", err))
 		}
 
+		gitAuthProviders := []*sdkproto.GitAuthProvider{}
+		for _, p := range templateVersion.GitAuthProviders {
+			link, err := server.Database.GetGitAuthLink(ctx, database.GetGitAuthLinkParams{
+				ProviderID: p,
+				UserID:     owner.ID,
+			})
+			if errors.Is(err, sql.ErrNoRows) {
+				continue
```

Comment:
I assumed that git auth was required if a template specifies it based on not being able to create a workspace in dogfood until I set it up. Should this not fail the build early or something?

Reply:
I think we might make this an option in the data source eventually, but the default should be to never fail builds.

This feature is mostly useful for a first-time clone anyways.